locale/en/knowledge/javascript-conventions/what-are-the-built-in-timer-functions.md

```diff
@@ -9,60 +9,95 @@ difficulty: 1
 layout: knowledge-post.hbs
 ---
 
-There are two built-in timer functions, `setTimeout` and `setInterval`, which can be used to call a function at a later time. For an example usage:
+There are two most common built-in timer functions, `setTimeout` and `setInterval`, which can be used to call a function at a later time. For an example usage:
 
-    setTimeout(function() { console.log("setTimeout: It's been one second!"); }, 1000);
-    setInterval(function() { console.log("setInterval: It's been one second!"); }, 1000);
+```js
+setTimeout(function() { console.log("setTimeout: It's been one second!"); }, 1000);
+setInterval(function() { console.log("setInterval: It's been one second!"); }, 1000);
+```
 
 An example output is:
 
-    setTimeout: It's been one second!
-    setInterval: It's been one second!
-    setInterval: It's been one second!
-    setInterval: It's been one second!
-    setInterval: It's been one second!
-    ...
+```bash
+setTimeout: It's been one second!
+setInterval: It's been one second!
+setInterval: It's been one second!
+setInterval: It's been one second!
+setInterval: It's been one second!
+...
+```
 
 As you can see the parameters to both are the same. The number second parameter says how long in milliseconds to wait before calling the function passed into the first parameter. The difference between the two functions is that `setTimeout` calls the callback only once while `setInterval` will call it over and over again.
 
 Typically you want to be careful with `setInterval` because it can cause some undesirable effects.  If, for example, you wanted to make sure your server was up by pinging it every second, you might think to try something like this:
 
-    setInterval(ping, 1000);
+```js
+setInterval(ping, 1000);
+```
 
 This can cause problems, however, if your server is slow and it takes, say, 3 seconds to respond to the first request. In the time it takes to get back the response, you would have sent off 3 more requests - not exactly desirable!  This isn't the end of the world when serving small static files, but if you doing an expensive operation such as database query or any complicated computation this can have some undesirable results. A common solution looks like this:
```

Comment:
```suggestion
This can cause problems, however, if your server is slow and it takes, for example, 3 seconds to respond to the first request. In the time it takes to get back the response, you would have sent off 3 more requests - not exactly desirable! Overall, this doesn't have a large impact when serving small static files. But if you're doing an expensive operation, such as a database query or any complex computation, this can have undesirable results. A common solution looks like this:
```